backend/typescript/services/implementations/campService.ts

```diff
@@ -1,17 +1,85 @@
-import { CreateCampDTO, CampDTO, CamperCSVInfoDTO } from "../../types";
+import {
+  CreateCampDTO,
+  CampDTO,
+  CamperCSVInfoDTO,
+  GetCampDTO,
+  FormQuestionDTO,
+} from "../../types";
 import ICampService from "../interfaces/campService";
 import MgCamp, { Camp } from "../../models/camp.model";
 import MgCamper, { Camper } from "../../models/camper.model";
 import { getErrorMessage } from "../../utilities/errorUtils";
 import { generateCSV } from "../../utilities/CSVUtils";
 import logger from "../../utilities/logger";
 import MgBaseCamp, { BaseCamp } from "../../models/baseCamp.model";
-import MgFormQuestion from "../../models/formQuestion.model";
+import MgFormQuestion, { FormQuestion } from "../../models/formQuestion.model";
 
 const Logger = logger(__filename);
 
 class CampService implements ICampService {
   /* eslint-disable class-methods-use-this */
+  async getCamps(): Promise<GetCampDTO[]> {
+    try {
+      const camps: Camp[] | null = await MgCamp.find({}).populate({
+        path: "baseCamp",
+        model: MgBaseCamp,
+        populate: {
+          path: "formQuestions",
+          model: MgFormQuestion,
+        },
+      });
+
+      if (!camps) {
+        return [];
+      }
+
+      return camps.map((camp) => {
+        const baseCamp = camp.baseCamp as BaseCamp;
+        const formQuestionArr = baseCamp.formQuestions as FormQuestion[];
+        const formQuestions = formQuestionArr.map(
+          (formQuestion: FormQuestion) => {
+            const {
+              id,
+              type,
+              question,
+              required,
+              description,
+              options,
+            } = formQuestion;
+            let result: FormQuestionDTO = { id, type, question, required };
+            if (description) {
+              result = { ...result, description };
+            }
+            if (options) {
+              result = { ...result, options };
+            }
```

Comment:
nice catch! makes much more sense

Reply:
I think that unpacking formQuestion results in mongoose fields being added: for eg. `"$__":{"strictMode":true,"selected":{},"getters":{},...` etc. Alternatively, we can just use this

`return formQuestion;` but this will result in __v and _id being added. For example:

`{
options: [],
_id: 6236dab5a84ccbbe42d813dd,
type: 'Multiselect',
question: 'Q1',
required: true,
description: 'question description',
__v: 0
}`